### PR TITLE
feat: adds copy deck functionality

### PR DESCRIPTION
**Pull request by blendaf**

### CODE REVIEW

File: components/deck/useEditDeck.js [hunk 1]
```diff
@@ -156,6 +156,20 @@ const useEditDeck = () => {
       })
       .catch(error => dispatch({ type: 'error', error }))
   }
+
+  const copyDeck = () => {
+    dispatch({ type: 'submit_changes' })
+    const deckObject = state.deck
```
Comment (juliuscc) [inline components/deck/useEditDeck.js]: I think we would want to update the createdAt field

Comment (juliuscc) [reply components/deck/useEditDeck.js]: Also, should the deck have the same name, or should it be called something like 'Copy of ...' like in google drive?

Comment (Falcasantos) [reply components/deck/useEditDeck.js]: We should definitely have unique names for decks imo, and we can append "copy" to the name of the deck, and if there are multiple copies, then we can name it "copy", "copy 2", "copy 3", etc. What do you think?

**Review by juliuscc — APPROVED** [review]

File: components/deck/useEditDeck.js (hunk 1 above)
Comment (juliuscc) [reply components/deck/useEditDeck.js]: Good idea!

For copy 1,2,3 we have to do an extra database call, and some extra logic, let’s do that as a separate pr.

**Pull request merged**